2021-04-30 11:36:12,870 : INFO : Start of Training
2021-04-30 11:36:12,870 : INFO : Run_id:2021-04-30_113612_237341765
2021-04-30 11:36:12,871 : INFO : Start of Data Load, validation and transformation
2021-04-30 11:36:12,871 : INFO : Start of Archiving Old Rejected Files...
2021-04-30 11:36:12,871 : INFO : End of Archiving Old Rejected Files...
2021-04-30 11:36:12,871 : INFO : Start of Archiving Old Validated Files...
2021-04-30 11:36:12,874 : INFO : End of Archiving Old Validated Files...
2021-04-30 11:36:12,874 : INFO : Start of Archiving Old Processed Files...
2021-04-30 11:36:12,874 : INFO : End of Archiving Old Processed Files...
2021-04-30 11:36:12,874 : INFO : Start of Archiving Old Result Files...
2021-04-30 11:36:12,875 : INFO : End of Archiving Old Result Files...
2021-04-30 11:36:12,875 : INFO : Start of Reading values From Schema...
2021-04-30 11:36:12,876 : INFO : End of Reading values From Schema...
2021-04-30 11:36:12,876 : INFO : Start of Validating Column Length...
2021-04-30 11:36:12,877 : INFO : End of Validating Column Length...
2021-04-30 11:36:12,877 : INFO : Start of Validating Missing Values...
2021-04-30 11:36:12,877 : INFO : End of Validating Missing Values...
2021-04-30 11:36:12,877 : INFO : Start of Replacing Missing Values with NULL...
2021-04-30 11:36:12,877 : INFO : End of Replacing Missing Values with NULL...
2021-04-30 11:36:12,877 : INFO : Start of Creating Table...
2021-04-30 11:36:12,886 : INFO : Opened training database successfully
2021-04-30 11:36:12,888 : INFO : Tables created successfully
2021-04-30 11:36:12,888 : INFO : Closed training database successfully
2021-04-30 11:36:12,888 : INFO : End of Creating Table...
2021-04-30 11:36:12,888 : INFO : Opened training database successfully
2021-04-30 11:36:12,889 : INFO : Start of Inserting Data into Table...
2021-04-30 11:36:12,889 : INFO : End of Inserting Data into Table...
2021-04-30 11:36:12,889 : INFO : Start of Exporting Data into CSV...
2021-04-30 11:36:12,889 : INFO : Opened training database successfully
2021-04-30 11:36:12,891 : INFO : End of Exporting Data into CSV...
2021-04-30 11:36:12,891 : INFO : Start of Moving Processed Files...
2021-04-30 11:36:12,892 : INFO : End of Moving Processed Files...
2021-04-30 11:36:12,892 : INFO : End of Data Load, validation and transformation
2021-04-30 11:36:12,892 : INFO : Start of Preprocessing...
2021-04-30 11:36:12,892 : INFO : Start of reading dataset...
2021-04-30 11:36:12,941 : INFO : End of reading dataset...
2021-04-30 11:36:12,942 : INFO : Start of Droping Columns...
2021-04-30 11:36:12,944 : INFO : End of Droping Columns...
2021-04-30 11:36:12,944 : INFO : Start of feature encoding...
2021-04-30 11:36:13,012 : INFO : End of feature encoding...
2021-04-30 11:36:13,013 : INFO : Start of Droping Columns...
2021-04-30 11:36:13,014 : INFO : End of Droping Columns...
2021-04-30 11:36:13,015 : INFO : Start of finding missing values...
2021-04-30 11:36:13,016 : INFO : End of finding missing values...
2021-04-30 11:36:13,016 : INFO : Start of splitting features and label ...
2021-04-30 11:36:13,018 : INFO : End of splitting features and label ...
2021-04-30 11:36:13,018 : INFO : End of Preprocessing...
2021-04-30 11:36:13,020 : INFO : Start of elbow plotting...
2021-04-30 11:36:13,021 : ERROR : Exception raised while elbow plotting:Found array with 0 sample(s) (shape=(0, 7)) while a minimum of 1 is required.
Traceback (most recent call last):
  File "D:\GitHub\Employee-Retention-Prediction\App\apps\tuning\cluster.py", line 27, in elbow_plot
    kmeans.fit(data) # fitting the data to the KMeans Algorithm
  File "D:\GitHub\Employee-Retention-Prediction\App\myenv\lib\site-packages\sklearn\cluster\_kmeans.py", line 858, in fit
    X = check_array(X, accept_sparse='csr', dtype=[np.float64, np.float32],
  File "D:\GitHub\Employee-Retention-Prediction\App\myenv\lib\site-packages\sklearn\utils\validation.py", line 583, in check_array
    raise ValueError("Found array with %d sample(s) (shape=%s) while a"
ValueError: Found array with 0 sample(s) (shape=(0, 7)) while a minimum of 1 is required.
2021-04-30 11:36:13,023 : ERROR : Unsuccessful End of Training
Traceback (most recent call last):
  File "D:\GitHub\Employee-Retention-Prediction\App\apps\tuning\cluster.py", line 27, in elbow_plot
    kmeans.fit(data) # fitting the data to the KMeans Algorithm
  File "D:\GitHub\Employee-Retention-Prediction\App\myenv\lib\site-packages\sklearn\cluster\_kmeans.py", line 858, in fit
    X = check_array(X, accept_sparse='csr', dtype=[np.float64, np.float32],
  File "D:\GitHub\Employee-Retention-Prediction\App\myenv\lib\site-packages\sklearn\utils\validation.py", line 583, in check_array
    raise ValueError("Found array with %d sample(s) (shape=%s) while a"
ValueError: Found array with 0 sample(s) (shape=(0, 7)) while a minimum of 1 is required.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "D:\GitHub\Employee-Retention-Prediction\App\apps\training\train_model.py", line 37, in training_model
    number_of_clusters = self.cluster.elbow_plot(self.X)
  File "D:\GitHub\Employee-Retention-Prediction\App\apps\tuning\cluster.py", line 43, in elbow_plot
    raise Exception()
Exception
